### PR TITLE
TODO JH: split /ui/assets into subpackages for each variant 

### PR DESCRIPTION
**Pull request by jhchabran**
TODO descr 

Duplicate /ui/assets into /ui/assets/[oss|enterprise] which enables to distinguish what needs to be built without relying on configuration flags, which in turn enables to not bust the analysis cache when building those. 

It's a bit crude as it relies on an anonmyous import to select the right bundle, but it's fairly localized and is good enough to hold until we stabilize how we handle assets in prod, where we'll be able to refactor this. 

TODO:

- doc stuff 
- ensure that app is working as intended in both prod / dev 
- wrap things up 
- reach out to app folks to ensure they're aware of the change. 

## Test plan

- `sg start oss-bazel` (need WEBPACK stuff commented out)

### CODE REVIEW

File: cmd/frontend/main.go [hunk 1]
```diff
@@ -2,10 +2,19 @@
...
 
 import (
+	"os"
+
 	"github.com/sourcegraph/sourcegraph/cmd/frontend/shared"
 	"github.com/sourcegraph/sourcegraph/cmd/sourcegraph-oss/osscmd"
+	"github.com/sourcegraph/sourcegraph/ui/assets"
+
+	// TODO
+	_ "github.com/sourcegraph/sourcegraph/ui/assets/oss"
```
Comment (jhchabran) [inline cmd/frontend/main.go:12]: that's where the trick is, as this tells gazelle which assets to properly embed.

File: cmd/frontend/main.go [hunk 2]
```diff
@@ -2,10 +2,19 @@
...
 )
 
 func main() {
+	if os.Getenv("WEBPACK_DEV_SERVER") == "1" {
```
Comment (jhchabran) [inline cmd/frontend/main.go:16]: meh but gotta get things working

File: ui/assets/dev.go [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
+	return LoadWebpackManifest()
+}
+
+func (p DevProvider) Assets() http.FileSystem {
+	return p.assets
+}
+
 var MockLoadWebpackManifest func() (*WebpackManifest, error)
 
 // LoadWebpackManifest uses Webpack manifest to extract hashed bundle names to
```
Comment (jhchabran) [inline ui/assets/dev.go:25]: I need to move this away, it's so easy to accidentally call this one and then have no assets in prod.

Comment (davejrt) [reply ui/assets/dev.go:25]: Can you explain how it would be easy to call this? I don't understand the issue

Comment (jhchabran) [reply ui/assets/dev.go]: Someone could just autocomplete on that package, see that function, the `LoadWebpackManifest` and assume it's the right one to use. That's why making it unexported fixes the problem.

**Review by davejrt — APPROVED** [review]

**Pull request closed without merging**